### DISCUSSION TITLE
How do i use ctx out side a command

### DISCUSSION THREAD

**Discussion by eclipsu** [Rapptz/discord.py#6616] [General]
How do i use ctx out side a command, i want to use ctx.author.id in my self reaction role command

**Comment by esmaycat** [comment]
`ctx` is a parameter that you can pass first into any command to give you access to the context of the message, where it was send etc... It can also be passed into commands inside cogs but the `self` parameter **must** come first. 

For your question itself, you cannot use it outside of a command as it is a special argument which you must put in the command method. If you wish to pass it into another method and call that method inside of the command and use it that way, that would still work however not in events. If you wanted to use it in an event: each event has its own parameters where data about the events go, if you look at those in the documentation (find the event reference [here](https://discordpy.readthedocs.io/en/latest/api.html#event-reference))  you will probably find what you want if need it for events

**Reply by eclipsu** [reply]
Thank you so much.
